### PR TITLE
FAH ensure secret

### PR DESCRIPTION
**Pull request by inlined**
Part two of secrets:set. Adds a utility to ensure a secret is created

### CODE REVIEW

File: src/apphosting/secrets/index.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      });
+      if (!stopTracking) {
+        return false;
+      }
+      delete existing.labels[FIREBASE_MANAGED];
+      await gcsm.patchSecret(project, secret, existing.labels);
+    }
+    // TODO: consider whether we should prompt a user who has an unmanaged secret to enroll in version control.
+    // This may not be a great idea until version control is actually implemented.
+  } catch (err: any) {
```
Comment (tonyjhuang) [inline src/apphosting/secrets/index.ts:50]: can we update this try catch to just wrap L20 to keep the try block focused? As it stands now, if `patchSecret` on L46 errors out for whatever reason, the user gets an unexpected error message

File: src/init/features/apphosting/secrets.ts [hunk 2]
```diff
@@ -73,6 +76,7 @@ export async function grantSecretAccess(
   }
 
   try {
+    // TODO: Merge with existing bindings with the same role
```
Comment (tonyjhuang) [inline src/init/features/apphosting/secrets.ts:79]: This is intentional, IAM will dedupe/coalesce roles fo rus

**Review by tonyjhuang — APPROVED** [review]
lgtm besides a couple nits, feel free to merge after addressing

File: src/init/features/apphosting/secrets.ts (hunk 2 above)
Comment (inlined) [reply src/init/features/apphosting/secrets.ts:79]: Would like to verify. If so, I believe this is new.

Comment (abhis3) [reply src/init/features/apphosting/secrets.ts:78]: I can confirm I verified this manually when testing the CLI

You can see that in the original version of the commit there was an `updateBindings` method https://github.com/firebase/firebase-tools/pull/6899/commits/0fb470c874e9c7a31c56d623af8195080f098909#diff-1de7c7b0fb2e58db23edd1a7905f884d38b83d93d51ed5edd39a0c082cc3de49L78-L98

**Pull request merged**